arangod/Aql/Function/ObjectFunctions.cpp

```diff
@@ -801,7 +801,12 @@ AqlValue functions::Entries(ExpressionContext* expressionContext,
   for (auto [key, value] : VPackObjectIterator(objectSlice, true)) {
     VPackArrayBuilder pair(builder.get(), true);
     builder->add(key);
-    builder->add(value);
+    if (value.isCustom()) {
+      builder->add(VPackValue(transaction::helpers::extractIdString(
+          trx->resolver(), value, objectSlice)));
+    } else {
+      builder->add(value);
+    }
```

Comment:
This is about handling the `_id` field, right? Is it possible for any other attribute to have custom value? WOuld it help to check the `key` for being `_id` instead of relying on the value to be custom?

Reply:
Only `_id` attributes can have custom values right now.
I would not tie this to the attribute name however, but rather to `velocypack::Slice::isCustom()`. The reason is that there may be `_id` attributes which are not of type custom (anymore), e.g. if they have already been extracted and turned into string value, or if the `_id` values were supplied by the user as part of the query and are not from a database document (and thus not custom).